submissions/alstep07/friends-app/script.js

```diff
@@ -0,0 +1,159 @@
+const MAX_FRIENDS_VALUE = 100;
+const main = document.querySelector(".main");
+const url = `https://randomuser.me/api/?results=${MAX_FRIENDS_VALUE}&noinfo`;
+const phoneImg = "./img/phone.png";
+const mailImg = "./img/mail.png";
+const errMessage = "server is not responding";
+const searchInput = document.querySelector("#search");
+const filterGender = document.querySelector(".search__by-gender");
+const filterName = document.querySelector(".search__by-alphabet");
+const filterAge = document.querySelector(".search__by-age");
+const filterDate = document.querySelector(".search__by-date");
+const resetBtn = document.querySelector(".search__reset__btn");
+const popupBtn = document.querySelector(".popup");
+const navbar = document.querySelector(".navbar");
+const filters = {
+	search: null,
+	gender: null,
+	age: null,
+	name: null,
+	date: null,
+};
+
+popupBtn.addEventListener("click", () => {
+	popupBtn.classList.toggle("popup-active");
+	navbar.classList.toggle("navbar-active");
+})
+
+const fetchFriends = async (url) => {
+	try {
+		let response = await fetch(url);
+		let data = await response.json();
+		return data.results;
+	} catch (err) {
+		main.innerHTML = `<div class="error">${errMessage}<br>(error: ${err})</div>`;
+	}
+};
+
+const init = (results) => {
+	render(results);
+	data = results.slice();
+
+	searchInput.addEventListener("keyup", ({ target }) => {
+		filters.search = target.value;
+		filterCards();
+	});
+
+	filterGender.addEventListener("click", ({ target }) => {
+		filters.gender = target.id;
+		filterCards();
+	});
+
+	filterName.addEventListener("click", ({ target }) => {
+		filters.name = target.id;
+		filters.age = null;
+		filters.date = null;
+		filterCards();
+	});
+
+	filterAge.addEventListener("click", ({ target }) => {
+		filters.age = target.id;
+		filters.name = null;
+		filters.date = null;
+		filterCards();
+	});
+
+	filterDate.addEventListener("click", ({ target }) => {
+		filters.date = target.id;
+		filters.age = null;
+		filters.name = null;
+		filterCards();
+	});
+
+	resetBtn.addEventListener("click", () => {
+		filters.search = null;
+		filters.gender = null;
+		filters.name = null;
+		filters.age = null;
+		render(results);
+	});
+};
+
+function filterCards() {
+	let filteredArr = data;
+	if (filters.search) filteredArr = sortBySearch(filteredArr, filters.search);
```

Comment:
Looks like a good case for switch statement. What do you think?

Reply:
You are right, i will change it.

Reply:
I have a problem with switch statement in this code snippet, because I can't apply more then one filter at same time, can I keep *if - else* statement there?

I tried this code: 

```javascript
switch (true) {
		case !!filters.search:
			filteredArr = sortBySearch(filteredArr, filters.search);
			break;
		case !!filters.gender:
			filteredArr = sortByGender(filteredArr, filters.gender);
			break;
		case !!filters.name:
			filteredArr = sortByName(filteredArr, filters.name);
			break;
		case !!filters.age:
			filteredArr = sortByAge(filteredArr, filters.age);
			break;
		case !!filters.date:
			filteredArr = sortByDate(filteredArr, filters.date);
			break;
	}
```

Reply:
Oh, yes, indeed, I missed that this way multiple filters will be impossible. Good catch!